File: Algoritmos_Ordenacion/Quick Sort/Quick.py
def Quick(arr):
    n = len(arr)
    if n <= 1:
        return arr

    # Elegimos el ultimo elemento como pivote
    pivot_index = n - 1
    pivot = arr[pivot_index]

    # Separamos los elementos menores y mayores que el pivote
    remaining_elements = arr[:pivot_index] + arr[pivot_index + 1:]

    less = [x for x in remaining_elements if x < pivot]
    greater = [x for x in remaining_elements if x >= pivot]
    return Quick (less) + [pivot] + Quick (greater)

# Sim
n = [45, -23, 78, 12, -56, 89, 34, -11, 67, 90,
     -78, 23, -45, 56, -89, -34, 11, -67, -90, 0,
     15, -32, 47, -61, 83, -5, 92, -74, 28, -39,
     51, -84, 63, -17, 76, -42, 95, -68, 7, -53,
     100, -99, 88, -77, 66, -55, 44, -33, 22, -1]
result = Quick (n)
print(result)

File: Algoritmos_Ordenacion/Quick Sort/Quick.java
public class Quick {
    static void qs(int[] a, int l, int r) {
        if (l >= r) return;
        int p = a[r], i = l - 1; //Comienza antes del primer elemento
        for (int j = l; j < r; j++)
            if (a[j] <= p) {
                i++;
                int t = a[i];
                a[i] = a[j];
                a[j] = t;
            }
        int t = a[i + 1]; a[i + 1] = a[r]; a[r] = t;
        qs(a, l, i);
        qs(a, i + 2, r);
    }

    public static void main(String[] args) {
        int[] a = {45, -23, 78, 12, -56, 89, 34, -11, 67, 90,
                -78, 23, -45, 56, -89, -34, 11, -67, -90, 0,
                15, -32, 47, -61, 83, -5, 92, -74, 28, -39,
                51, -84, 63, -17, 76, -42, 95, -68, 7, -53,
                100, -99, 88, -77, 66, -55, 44, -33, 22, -1};
        qs(a, 0, a.length - 1);
        for (int n : a) System.out.print(n + " ");
    }
}